### ISSUE TITLE
Import/export app settings

### ISSUE THREAD

**Issue by DomeDrake** [orgzly-revived/orgzly-android-revived#128]
- I have searched for existing issues that may be the same as or related to mine.

I would love to see this feature added. Not only would it greatly enhance the ease of setup of Orgzly R on a new phone, it would also give great peace of mind knowing one has not forgotten to write down anything when documenting ones own settings.

Thank you for all the work! :)

For the sake of completeness: This has been suggested for the original App [orgzly/orgzly-android #156](https://github.com/orgzly/orgzly-android/issues/156) by @mjf, too.

**Comment by stfl** [comment]
+1

**Comment by amberin** [comment]
Yes, I believe we should add an option to store all settings in a hidden file or dir in a repo. Either in a single JSON/TOML/YAML file, or in a .orgzlysettings directory with multiple files.

This should be a big UX improvement when it comes to saving search queries and any other settings.

The configuration scheme would probably need to be versioned to allow improvements without breaking stuff.

I'm thinking there would be a new section somewhere in settings, where the user can choose which settings they want to store in the repo, if any. Or maybe that's over-engineering things, and we should start with a simple toggle: either store all settings in the file, or none?

I suppose it would be useless (or risky) to store configured repos in this file, even though we do support multiple repos, and the file would only go into one of them.

Currently, Git repos support an .orgzlyignore file. This should probably be supported by all repos (as a general solution to things like #147), and moved to this solution, if implemented.

**Comment by DanielKonopka** [comment]
Why not simply allow for importing & exporting of the entire /data/data/com.orgzlyrevived/databases/orgzly.db SQLite file?

**Comment by amberin** [comment]
> Why not simply allow for importing & exporting of the entire /data/data/com.orgzlyrevived/databases/orgzly.db SQLite file?

Sure, that's possible, but a text file would be much more portable.

I also think a major benefit of storing settings in a text file in the repo would always permanent backup of settings, so that setting up a new device is always really easy, whether you bothered to make exports or not.

**Comment by Ypot** [comment]
A headline in any of the orgzly notebooks would do it?

Like:
** Orgzly revived settings
...

**Comment by amberin** [comment]
Yes, that is definitely worth looking into. We already have the ability to parse and write ORG files, but not other file types, so an ORG-based solution may very well require less new code.

**Comment by amberin** [comment]
I believe we should start with a manual export/import feature. Continuously syncing to a repo file is a bit more involved, since it involves changes and conflicts.

**Comment by DanielKonopka** [comment]
@amberin Any heads-up on this? It's an absolutely crucial feature. Wish I could implement this on my own and contribute that way.

**Comment by rezaamashi** [comment]
+1 on this. This has been a hassle for me anytime I update my OS or for some reason I had to reinstall Orgzly Revived. 

I have my preferred setup and some custom searches for widgets. Going back to the docs and setting it up again one by one every time things come up is a lot of hassle in the UX.
